### ISSUE TITLE
Ignoring URL path when config.locationType is set to "hash"

### ISSUE THREAD

**Issue by chrisdpeters** [ember-cli/ember-cli#3566]
So far, I've wanted the behavior of my app to be such, which I understand to be off the beaten path:

-  Hash-based routing (i.e., `config.locationType = "hash"`)
-  Ignore the path part of the URL and serve `app/index.html` every time.

To get the 2nd part working, I had to create a (ridiculous?) add-on called ember-cli-hash-anywhere:
https://github.com/liveeditor/ember-cli-hash-anywhere

It's really just a copy-and-paste of ember-cli's `history-support-middleware` with the `if` statement [switched out](https://github.com/liveeditor/ember-cli-hash-anywhere/blob/master/index.js#L20) to run for a `locationType` of `hash` instead of `auto` or `history`.

Do you think it would make sense to add a flag or something to Ember CLI to allow this behavior? Again, I understand I am in the minority for wanting it to begin with and can just maintain the add-on for my purposes if need be.

**Comment by davidpett** [comment]
@chrisdpeters question about this, did you run `ember serve` before changing the `locationType` to hash? I have a similar situation, but if you want to make it work, just change it in your `router.js` file to `'hash'` instead of the `config.locationType` i'm trying to nail down the reason this is happening

**Comment by chrisdpeters** [comment]
@davidpett I just tried these steps:

1. Stop the ember server.

2. Remove my ember-cli-hash-anywhere add-on from `package.json` and run `npm install`.

3. Change `app/router.js` to look like this:
    ```js
    import Ember from 'ember';

    var Router = Ember.Router.extend({
      location: 'hash'
    });

    export default Router.map(function() {
      this.route('editor');
    });
    ```
4.  Start ember server.

When I browse to `/about`, I still get a 404. When I add my add-on back, it works again.

**Comment by chrisdpeters** [comment]
I created a blank project with Ember CLI 0.2.5, and this appears to not be an issue anymore.

In the meantime, I ended up implementing a solution based on a Dockyard guy's [ember-cli-plus-backend](https://github.com/danmcclain/ember-cli-plus-backend) that works better for my situation.

But anyway, anyone using a plain vanilla Ember CLI app without backend trickery shouldn't be running into this issue anymore.
